services/upload/src/main/java/app/coronawarn/server/services/federation/upload/runner/TestDataGeneration.java

```diff
@@ -125,36 +100,28 @@ private long secondsToHours(long timestampInSeconds) {
           hourStart,
           hourEnd);
       return IntStream.range(0, numberOfKeysToGenerate)
-          .mapToObj(ignoredValue -> this.makeKeyFromTimestamp(this.getRandomBetween(hourStart, hourEnd)))
+          .mapToObj(ignoredValue -> generateDiagnosisKey(getRandomBetween(hourStart, hourEnd), "DE"))
```

Comment:
I wonder if we are going to run into problems in the future because this was changed to `DiagnosisKey` rather than `FederationUploadKey`. As of now it seems to be all good because `FederationUploadKey` is a superset of `DiagnosisKey`. And even in the context of this class we are not using any specific properties of the Federation model. The only thing I can imagine is that if in the future we have multiple EFGS backends, we would need to refactor this again because TestDataGeneration will have to insert on Federation specific columns.

If you don't see any complications in this code to refactor now, I'd suggest you do it. Otherwise we can just wait and see what happens and change this only if needed.

Reply:
Good point! I've been thinking about this approach. Please let me know about the new changes.